test/test_reductions.py

```diff
@@ -2992,6 +2999,46 @@ def test_reduction_empty_any_all(self, device):
             self.assertEqual(torch.ones((2, 1, 4), device=device, dtype=out_dtype), xb.all(1, keepdim=True))
             self.assertEqual(torch.ones((), device=device, dtype=out_dtype), xb.all())
 
+    @ops(reference_masked_ops)
+    def test_reference_masked(self, device, dtype, op):
+        # Typically, op.ref is a reference function using numpy
```

Comment:
I think our semantics agree with [MaskedArray](https://numpy.org/doc/stable/reference/generated/numpy.ma.sum.html) though (aside from the inverted mask), right?

Reply:
I'm also wondering whether it makes sense to add an autograd test, in particular around things like a fully masked out row in a 2-D Tensor. What will be the gradient? This will matter once we write one-off kernels and there might be some subtleties to consider here.

Reply:
> I think our semantics agree with [MaskedArray](https://numpy.org/doc/stable/reference/generated/numpy.ma.sum.html) though (aside from the inverted mask), right?

Yes, I think this is true and my current plan is to use numpy.ma tools for reference as implementing the reference from scratch is not trivial considering possible nan/inf inputs.

Reply:
Now testing masked reductions on strided tensors use numpy reduction operations.